### DISCUSSION TITLE
谷歌翻译网络错误的解决办法（已失效）

### DISCUSSION THREAD

**Discussion by YiiGuxing** [YiiGuxing/TranslationPlugin#2114] [Q&A] [answered]
近日使用谷歌翻译和语音朗读时会出现网络错误的问题，如何解决？

**Comment by YiiGuxing** [comment, marked as answer]
<br/>

## 注意：目前此方法已经失效！！！
还想继续使用谷歌翻译目前唯一的办法就只有使用梯子了，而没有梯子的就只能先使用其他的翻译引擎了，至于语音朗读，再想办法吧，暂时先这样。

> 提示：使用梯子的同学别忘了把之前添加到Hosts文件的 IP-域名映射删除掉哦。

<br/>
<br/>

_原内容：_


## 问题原因
疑是谷歌关停了国内的翻译服务，导致相关API无法使用。

## 解决办法
> _感谢 @zhangkaiqiu 在 https://github.com/YiiGuxing/TranslationPlugin/issues/2051#issuecomment-1261880391 中提供的解决办法！_

如果您有梯子，您可以通过梯子正常使用。如果没有，则可将以下内容添加到 hosts 文件中以暂时解决此问题，恢复谷歌翻译和语音朗读功能的正常使用：

```
203.208.40.66  translate.googleapis.com
``` 

### 1. Windows 系统
[点击这里](https://github.com/YiiGuxing/TranslationPlugin/files/9757606/fix_google_translate.zip)下载并解压缩后得到一个脚本文件：`fix_google_translate.bat`，鼠标右键“**以管理员身份运行**”即可。

### 2. macOS
在终端执行以下命令：
```shell
sudo echo -e '\n\n203.208.40.66  translate.googleapis.com' >> /etc/hosts
sudo killall -HUP mDNSResponder
```

### 3. Linux 系统
在终端执行以下命令：
```shell
sudo echo -e '\n\n203.208.40.66  translate.googleapis.com' >> /etc/hosts
sudo nscd restart
```

> Tips： 上述方法执行后仍无效可以尝试重启机器。

## 补充
如果上面提供的IP不可用，访问[此网站](https://ping.chinaz.com/translate.google.cn)可以获取其他的IP，选择国内离自己位置较近或响应时间较短的IP即可（国外IP可能是不可用的）。

- _Windows 环境的朋友可以使用记事本打开上面提供的脚本编辑 IP。_

_注意：应用新IP之前要将之前设置的删除掉，否则新设置的将无效。_
> _感谢 @cuizhixuan 在 https://github.com/YiiGuxing/TranslationPlugin/issues/2051#issuecomment-1279677885 中提供的此方法。_

**Comment by 1574596522** [comment]
试了半天才知道要在上面加上“#gugle”
# guogle
203.208.40.66 translate.google.com
203.208.40.66 translate.googleapis.com

**Reply by YiiGuxing** [reply]
???

不加也可以啊。。。
